src/payment/strategies/cardinal/cardinal-client.ts

```diff
@@ -37,6 +37,19 @@ export default class CardinalClient {
         return this._sdk.then(() => {});
     }
 
+    isConfigured(): boolean {
+        return this._isConfigured;
+    }
+
+    async reconfigure(provider: string, clientToken: string, testMode?: boolean): Promise<void> {
+        this._sdk = undefined;
```

Comment:
I'm wondering if it might be a better idea to re-use the `configure` method and instead of storing `isConfigured: bool`, store `configurationToken: string`

When trying to call configure() again with a different token to the one being configured, it can re-configure. This would simplifiy the logic for clients and avoid leaking internal logic in `cardinal-three-d-secure-flow-v2.ts `

Reply:
Sounds good! 👍